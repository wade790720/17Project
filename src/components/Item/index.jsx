import React from 'react';
import cx from "classnames";
import PropTypes from 'prop-types';

import "./style/item.sass"

class Item extends React.Component {
    state = { hasFocus: false }

    toggle = () => {
        this.setState({ hasFocus: !this.state.hasFocus });
    }

    handleChange = (target, e) => {
        let data = { ...this.props.item };
        data = {
            ...data,
            [target]: e.target.value
        };
        this.props.onChange(data);
    }

    render() {
        const { item, disabled } = this.props;
        return (
            <div className={cx("item", { "focus": this.state.hasFocus }, { "disabled": disabled })}>
                <input type="text" value={item.name} onChange={e => this.handleChange("name", e)} onFocus={this.toggle} onBlur={this.toggle} placeholder="Name" disabled={disabled} />
                <input type="number" value={item.price} onChange={e => this.handleChange("price", e)} onFocus={this.toggle} onBlur={this.toggle} placeholder="Price" disabled={disabled} />
                <input type="text" value={item.remark} onChange={e => this.handleChange("remark", e)} onFocus={this.toggle} onBlur={this.toggle} placeholder="Remark" disabled={disabled} />
                <div className="remove" onClick={this.props.onDelete} />
            </div>
        );
    }

};

export default Item;

Item.defaultProps = {
    item: {},
    disabled: false,
    onChange: () => { },
    onDelete: () => { }
}
Item.propTypes = {
    item: PropTypes.object,
    disabled: PropTypes.bool,
    onChange: PropTypes.func,
    onDelete: PropTypes.func
}