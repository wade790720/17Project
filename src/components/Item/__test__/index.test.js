import React from "react";
import { shallow } from "enzyme";
import Item from "../index";
import renderer from "react-test-renderer";

describe('Item tests', () => {
    it("Default view", () => {
        const tree = renderer.create(
            <Item />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("toggle function", () => {
        const component = shallow(<Item />);
        component.instance().toggle();
        expect(component.instance().state.hasFocus).toBeTruthy();
        component.instance().toggle();
        expect(component.instance().state.hasFocus).toBeFalsy();
    });
});