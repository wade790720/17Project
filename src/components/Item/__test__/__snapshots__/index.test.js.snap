// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item tests Default view 1`] = `
<div
  className="item"
>
  <input
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Name"
    type="text"
  />
  <input
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Price"
    type="number"
  />
  <input
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Remark"
    type="text"
  />
  <div
    className="remove"
    onClick={[Function]}
  />
</div>
`;
